synapse/third_party_rules/access_rules.py

```diff
@@ -442,14 +476,24 @@ def _on_membership_or_invite_restricted(self, event: EventBase) -> bool:
         invitee_domain = get_domain_from_id(event.state_key)
         return invitee_domain not in self.domains_forbidden_when_restricted
 
-    def _on_membership_or_invite_unrestricted(self) -> bool:
+    def _on_membership_or_invite_unrestricted(
+        self, event: EventBase, state_events: StateMap[EventBase]
+    ) -> bool:
         """Implements the checks and behaviour specified for the "unrestricted" rule.
 
-        "unrestricted" currently means that every event is allowed.
```

Comment:
I don't think we should be removing this line - instead we should fix it with the new behaviour to follow for membership events in unrestricted rooms.